src/terminal/terminal/terminal.h

```diff
@@ -492,6 +492,14 @@ int guac_terminal_render_frame(guac_terminal* terminal);
 int guac_terminal_read_stdin(guac_terminal* terminal, char* c, int size);
 
 /**
+ * Manually stop the terminal to forcibly unblock any pending reads/writes,
+ * e.g. forcing guac_terminal_read_stdin() to return and cease all terminal I/O.
+ * This function has been called in guac_terminal_free(). It could be useful
```

Comment:
> This function has been called in guac_terminal_free().

I would suggest leaving this out as an internal implementation detail. If the intent is to document that `guac_terminal_free()` will also unblock pending reads/writes, then that information should be added to the documentation of `guac_terminal_free()`.

> It could be useful when need to stop the terminal before calling guac_terminal_free().

This doesn't add any information. Really, everything you had prior to this last sentence was perfectly fine. If the caller needs to invoke this function, the first couple of lines documenting that the function forcibly unblocks pending reads/writes are exactly what is needed.